app/controllers/orgs/rosters_controller.rb

```diff
@@ -228,6 +311,88 @@ def get_user_to_group_hash(grouping)
 
       mapping
     end
+
+    # Returns name of the linked google course to current organization (for syncing rosters)
+    def current_organization_google_course_name
+      return nil if current_organization.google_course_id.nil?
+      authorize_google_classroom
+      course = @google_classroom_service.get_course(current_organization.google_course_id)
+      return course.name unless course.nil?
+      nil
+    rescue Google::Apis::Error
+      nil
+    end
+
+    # Returns list of students in a google classroom with error checking
+    # rubocop:disable Metrics/MethodLength
+    def list_google_classroom_students(course_id)
+      response = @google_classroom_service.list_course_students(course_id)
+      response.students ||= [] # Set to empty array if no students in course
+    rescue Google::Apis::AuthorizationError
+      google_classroom_client = GitHubClassroom.google_classroom_client
+      login_hint = current_user.github_user.login
+      redirect_to google_classroom_client.get_authorization_url(login_hint: login_hint, request: request)
+      nil
+    rescue Google::Apis::ServerError, Google::Apis::ClientError
+      flash[:error] = "Failed to fetch students from Google Classroom. Please try again later."
+      redirect_to organization_path(current_organization)
+      nil
+    end
+    # rubocop:enable Metrics/MethodLength
+
+    # Add Google Classroom students to roster
+    def add_google_classroom_students(students)
+      names = students.map { |s| s.profile.name.full_name }
+      user_ids = students.map(&:user_id)
+      params[:identifiers] = names.join("\r\n")
+      params[:google_user_ids] = user_ids
+
+      if current_roster.nil?
+        create
```

Comment:
This pattern of calling controller actions from within controller actions can lead to some weird side effects. Can we abstract the logic you need into a method and then call that method from both controller actions?